src/libraries/System.Net.Quic/tests/FunctionalTests/QuicConnectionTests.cs

```diff
@@ -37,23 +37,141 @@ public async Task TestConnect()
+
+                    sync.Release();
+
+                    // Pending ops should fail
+                    await Assert.ThrowsAsync<QuicOperationAbortedException>(() => acceptTask);
+                    await Assert.ThrowsAsync<QuicOperationAbortedException>(() => connectTask);
+
+                    // Subsequent attempts should fail
+                    // TODO: Should these be QuicOperationAbortedException, to match above? Or vice-versa?
+                    await Assert.ThrowsAsync<ObjectDisposedException>(async () => await serverConnection.AcceptStreamAsync());
+                    await Assert.ThrowsAsync<ObjectDisposedException>(async () => await OpenAndUseStreamAsync(serverConnection));
+                });
+        }
+
+        [Fact]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/55242", TestPlatforms.Linux)]
+        public async Task ConnectionClosedByPeer_WithPendingAcceptAndConnect_PendingAndSubsequentThrowConnectionAbortedException()
+        {
+            if (IsMockProvider)
+            {
+                return;
+            }
+
+            using var sync = new SemaphoreSlim(0);
+
+            await RunClientServer(
+                async clientConnection =>
                 {
                     await sync.WaitAsync();
-                    QuicConnectionAbortedException ex = await Assert.ThrowsAsync<QuicConnectionAbortedException>(() => serverConnection.AcceptStreamAsync().AsTask());
+
+                    await clientConnection.CloseAsync(ExpectedErrorCode);
+                },
+                async serverConnection =>
+                {
+                    // Pend operations before the client closes.
+                    Task<QuicStream> acceptTask = serverConnection.AcceptStreamAsync().AsTask();
+                    Assert.False(acceptTask.IsCompleted);
+                    Task<QuicStream> connectTask = OpenAndUseStreamAsync(serverConnection);
+                    Assert.False(acceptTask.IsCompleted);
+
+                    sync.Release();
+
+                    // Pending ops should fail
+                    QuicConnectionAbortedException ex;
+
+                    ex = await Assert.ThrowsAsync<QuicConnectionAbortedException>(() => acceptTask);
+                    Assert.Equal(ExpectedErrorCode, ex.ErrorCode);
+                    ex = await Assert.ThrowsAsync<QuicConnectionAbortedException>(() => connectTask);
+                    Assert.Equal(ExpectedErrorCode, ex.ErrorCode);
+
+                    // Subsequent attempts should fail
+                    ex = await Assert.ThrowsAsync<QuicConnectionAbortedException>(() => serverConnection.AcceptStreamAsync().AsTask());
                     Assert.Equal(ExpectedErrorCode, ex.ErrorCode);
+                    // TODO: ActiveIssue https://github.com/dotnet/runtime/issues/56133
+                    // MsQuic fails with System.Net.Quic.QuicException: Failed to open stream to peer. Error Code: INVALID_STATE
```

Comment:
Can we at least check that there's is *an* exception on MsQuic and put the check of the exact exception under active issue?

Reply:
Done